portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -37,8 +55,25 @@ public void init() {
 
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    
+    Query query = new Query("Comment").addSort("timestamp", SortDirection.DESCENDING);
+
+    DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+    PreparedQuery results = datastore.prepare(query);
 
-    String json = convertToJson(comments);
+    for (Entity entity : results.asIterable()) {
+      String name = (String) entity.getProperty("name");
+      String text = (String) entity.getProperty("comment");
```

Comment:
What happens if the property isn't set? Or is it guaranteed to alway shave a property called "name"?

Reply:
In doPost, every property is set before it is stored in Datastore, and so I am thinking the code should not run into this issue. What do you think?

Reply:
What if, in the future, you add a new property that old data doesn't have?